### PR TITLE
Feature/farm 63 view profile

### PR DESCRIPTION
**Pull request by jessicahernandezhv**
[[FARM-63 - View Profile](https://amidodevelopment.atlassian.net/secure/RapidBoard.jspa?rapidView=150&projectKey=FARM&view=planning&selectedIssue=FARM-63)]

#### 📲 What

Builded user profile UI

#### 🤔 Why
		
Because we need to save and show the user profile details.
		
#### 🛠 How
		
Applying the new code Style for Flutter

#### ✅ Acceptance criteria

- [ ] Design Review for UI with BA completed. 
- [x] Manually tested and verified on Android device/emulator (min. Lollipop 5.1)
- [x] Showcase video / automation test
- [ ] Passing all tests
- [x] Rebased/merged with latest changes from development and re-tested
- [x] Removed unsed comments. TODOs must have JIRA for future work.

#### Coding Standards Checklist
- [ ] unit tests 80% coverage on testable code (functions, methods, class)
- [ ] (Architecture) redux state, reducers, actions, middleware defined under lib/redux
- [ ] ui elements defined under lib/ui. 
- [ ] Strings ready for localisation (e.g. defined in lib/utils/strings.dart)
- [ ] Assets (images/icons path) defined in lib/utils/assets.dart
- [ ] Colors defined in lib/utils/colors.dart
- [ ] Dimensions (ie margins padding) defined in lib/utils/dimens.dart
- [ ] TextStyles defined in lib/utils/styles.dart

Recommended Style Guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo

#### Reviewers

### CODE REVIEW

File: lib/ui/common/ActionSheet.dart [hunk 1]
```diff
@@ -115,6 +116,14 @@ class ActionSheet extends StatefulWidget {
 
   @override
   _ActionSheetState createState() => _ActionSheetState();
+
+  //FIXME: To show ActionBottomSheet (need to implement)
```
Comment (LeeMatthewHiggins) [inline lib/ui/common/ActionSheet.dart]: what's this?

File: lib/ui/profile/UserProfile.dart [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+import 'package:intl/intl.dart';
+
+class _Strings {
+  static final String ACTIVE_CROPS = Intl.message("Active crops");
+  static final String COMPLETED = Intl.message("Completed");
+  static final String SWITCH_PROFILE = Intl.message("Switch Profile");
+}
+
+class UserProfileViewModel {
+  List<UserProfileListItemViewModel> actions;
```
Comment (LeeMatthewHiggins) [inline lib/ui/profile/UserProfile.dart]: maybe call these "items"

File: lib/ui/profile/UserProfile.dart [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+  String userName;
```
Comment (LeeMatthewHiggins) [inline lib/ui/profile/UserProfile.dart]: small typo username

File: lib/ui/profile/UserProfile.dart [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+            ],
+          ),
+        ),
+        Padding(
+          padding: _style.edgePadding,
+          child: Column(
+            children: <Widget>[
+              RoundedButton(
+                viewModel: RoundedButtonViewModel(
+                  title: _Strings.SWITCH_PROFILE,
```
Comment (LeeMatthewHiggins) [inline lib/ui/profile/UserProfile.dart]: no snake case... camel case please. (search project for _Strings classes and see)

File: lib/ui/profile/UserProfile.dart [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+              ),
+            ],
+          ),
+        ),
+        SizedBox(height: _style.buttonLineSpace),
+      ],
+    );
+  }
+
+  Widget _buildListOfActions() {
```
Comment (LeeMatthewHiggins) [inline lib/ui/profile/UserProfile.dart]: _buildItemList() maybe better.

File: lib/ui/profile/UserProfileListItem.dart [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+    UserProfileListItemViewModel viewModel,
+    UserProfileListItemStyle style = _defaultStyle,
+  })  : this._viewModel = viewModel,
+        this._style = style,
+        super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return ListTile(
+      onTap: () => _viewModel.onTap(_viewModel.title),
```
Comment (LeeMatthewHiggins) [inline lib/ui/profile/UserProfileListItem.dart]: why pass title?

**Review by LeeMatthewHiggins — COMMENTED** [review]
Good so far, just some minor comments

File: lib/ui/profile/UserProfile.dart (hunk 3 above)
Comment (jessicahernandezhv) [reply lib/ui/profile/UserProfile.dart]: Done

File: lib/ui/profile/UserProfile.dart (hunk 2 above)
Comment (jessicahernandezhv) [reply lib/ui/profile/UserProfile.dart]: Done

File: lib/ui/profile/UserProfile.dart (hunk 5 above)
Comment (jessicahernandezhv) [reply lib/ui/profile/UserProfile.dart]: Done

File: lib/ui/common/ActionSheet.dart (hunk 1 above)
Comment (jessicahernandezhv) [reply lib/ui/common/ActionSheet.dart]: I think I added a blank line in a file outside of my PR.

File: lib/ui/profile/UserProfile.dart (hunk 4 above)
Comment (jessicahernandezhv) [reply lib/ui/profile/UserProfile.dart]: Changed

File: lib/ui/profile/UserProfileListItem.dart (hunk 6 above)
Comment (jessicahernandezhv) [reply lib/ui/profile/UserProfileListItem.dart]: Is not necessary. I applied just to print the selected element.

File: lib/redux/app/app_reducer.dart [hunk 7]
```diff
@@ -1,7 +1,6 @@
 import 'package:farmsmart_flutter/redux/app/app_state.dart';
 import 'package:farmsmart_flutter/redux/home/home_reducer.dart';
 
-AppState appReducer(AppState state, dynamic action) =>
-    AppState(
-        homeState: homeReducer(state.homeState ,action),
-    );
\ No newline at end of file
+AppState appReducer(AppState state, dynamic action) => AppState(
```
Comment (LeeMatthewHiggins) [inline lib/redux/app/app_reducer.dart]: is this needed for PR?

File: lib/ui/profile/UserProfile.dart [hunk 8]
```diff
@@ -0,0 +1,353 @@
+import 'package:farmsmart_flutter/ui/common/ActionSheet.dart';
+import 'package:farmsmart_flutter/ui/common/ListDivider.dart';
+import 'package:farmsmart_flutter/ui/common/roundedButton.dart';
+import 'package:farmsmart_flutter/ui/profile/UserProfileListItem.dart';
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+
+class _Strings {
+  static final String activeCrops = Intl.message("Active crops");
```
Comment (LeeMatthewHiggins) [inline lib/ui/profile/UserProfile.dart]: this will not work....     If I change the language I will get the language that was set on app startup.....
have the string g only here and the Intl.message(_String.mystring) in the widget

**Review by LeeMatthewHiggins — COMMENTED** [review]
Just the localised strings issue
Good work 👍

File: lib/redux/app/app_reducer.dart (hunk 7 above)
Comment (jessicahernandezhv) [reply lib/redux/app/app_reducer.dart]: Android Studio was highlighting this file with an error (don't know why), that's why I reformat the code, but didn't make any changes.

Comment (davidfranquet) [inline lib/redux/app/app_reducer.dart]: Revert this change. not part of this PR

File: lib/ui/common/ActionSheet.dart [hunk 9]
```diff
@@ -117,6 +118,14 @@ class ActionSheet extends StatefulWidget {
...
+  static Future onMenuPressed(BuildContext context) async {
+    showModalBottomSheet(
+        context: context,
+        builder: (widgetBuilder) =>  ActionSheet(viewModel: MockActionSheetViewModel.buildWithCheckBox(), style: ActionSheetStyle.defaultStyle())
```
Comment (davidfranquet) [inline lib/ui/common/ActionSheet.dart]: Format code
```suggestion
        builder: (widgetBuilder) =>  ActionSheet(viewModel: MockActionSheetViewModel.buildWithCheckBox(), style: ActionSheetStyle.defaultStyle(),)
```

File: lib/ui/profile/UserProfile.dart (hunk 8 above)
Comment (jessicahernandezhv) [reply lib/ui/profile/UserProfile.dart]: Done

File: lib/ui/profile/UserProfile.dart [hunk 10]
```diff
@@ -0,0 +1,353 @@
...
+  final EdgeInsets headerEdgePadding;
+  final EdgeInsets titleLineSpace;
+
+  final BorderRadius buttonBorderShape;
+
+  final double imageSpacing;
+  final double subtitleLineSpace;
+  final double buttonLineSpace;
+  final double detailSpacing;
+  final double dividerHeight;
```
Comment (davidfranquet) [inline lib/ui/profile/UserProfile.dart]: Dividers, paddings, line spaces, divider color, etc... Should go on `_Constants` we will never change this. Just add the params that you think we want to change, for example: textstyles, button colors, max lines, image size, button size... 😄

File: lib/ui/mockData/MockUserProfileViewModel.dart [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+    );
+  }
+
+  static _mockItemTap() {
+    print("Was tapped");
+  }
+}
+
+
+List<String> _mockActionTitle = [
```
Comment (davidfranquet) [inline lib/ui/mockData/MockUserProfileViewModel.dart:83]: Test large titles

File: lib/ui/mockData/MockUserProfileViewModel.dart [hunk 12]
```diff
@@ -0,0 +1,80 @@
+import 'dart:math';
+
+import 'package:farmsmart_flutter/ui/profile/UserProfile.dart';
+import 'package:farmsmart_flutter/ui/profile/UserProfileListItem.dart';
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+import 'package:farmsmart_flutter/data/model/mock/MockString.dart';
+
+class MockUserProfileViewModel {
+  static UserProfileViewModel build() {
```
Comment (davidfranquet) [inline lib/ui/mockData/MockUserProfileViewModel.dart:10]: add another build with large titles and add it to playground

**Review by davidfranquet — COMMENTED** [review]
Looks good 👍  just minor comments

File: lib/ui/mockData/MockUserProfileViewModel.dart (hunk 11 above)
Comment (jessicahernandezhv) [reply lib/ui/mockData/MockUserProfileViewModel.dart:83]: Done

File: lib/ui/mockData/MockUserProfileViewModel.dart (hunk 12 above)
Comment (jessicahernandezhv) [reply lib/ui/mockData/MockUserProfileViewModel.dart:10]: Done

File: lib/ui/profile/UserProfile.dart (hunk 10 above)
Comment (jessicahernandezhv) [reply lib/ui/profile/UserProfile.dart]: Done

**Review by LeeMatthewHiggins — COMMENTED** [review]
Just davids comments, good to go when fixe, good work

**Review by LeeMatthewHiggins — DISMISSED** [review]
LGTM 👍

**Review by davidfranquet — APPROVED** [review]

**Review by LeeMatthewHiggins — APPROVED** [review]
LGTM 👍
